ibm/service/vpc/resource_ibm_is_share_mount_target.go

```diff
@@ -0,0 +1,328 @@
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The type of resource referenced.",
+			},
+		},
+	}
+}
+
+func ResourceIBMIsShareMountTargetValidator() *validate.ResourceValidator {
+	validateSchema := make([]validate.ValidateSchema, 1)
+	validateSchema = append(validateSchema,
+		validate.ValidateSchema{
+			Identifier:                 "name",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Optional:                   true,
+			Regexp:                     `^([a-z]|[a-z][-a-z0-9]*[a-z0-9]|[0-9][-a-z0-9]*([a-z]|[-a-z][-a-z0-9]*[a-z0-9]))$`,
+			MinValueLength:             1,
+			MaxValueLength:             63,
+		},
+	)
+
+	resourceValidator := validate.ResourceValidator{ResourceName: "ibm_is_share_mount_target", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIBMIsShareMountTargetCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcClient, err := meta.(conns.ClientSession).VpcV1BetaAPI()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createShareMountTargetOptions := &vpcbetav1.CreateShareMountTargetOptions{}
+
+	createShareMountTargetOptions.SetShareID(d.Get("share").(string))
+	vpcid := d.Get("vpc").(string)
+	vpc := &vpcbetav1.VPCIdentity{
+		ID: &vpcid,
+	}
+	createShareMountTargetOptions.SetVPC(vpc)
+	if _, ok := d.GetOk("name"); ok {
+		createShareMountTargetOptions.SetName(d.Get("name").(string))
+	}
+
+	shareTarget, response, err := vpcClient.CreateShareMountTargetWithContext(context, createShareMountTargetOptions)
+	if err != nil {
+		log.Printf("[DEBUG] CreateShareMountTargetWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+	_, err = WaitForTargetAvailable(context, vpcClient, *createShareMountTargetOptions.ShareID, *shareTarget.ID, d, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", *createShareMountTargetOptions.ShareID, *shareTarget.ID))
+	d.Set("mount_target", *shareTarget.ID)
+	return resourceIBMIsShareMountTargetRead(context, d, meta)
+}
+
+func resourceIBMIsShareMountTargetRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcClient, err := meta.(conns.ClientSession).VpcV1BetaAPI()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getShareMountTargetOptions := &vpcbetav1.GetShareMountTargetOptions{}
+
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getShareMountTargetOptions.SetShareID(parts[0])
+	getShareMountTargetOptions.SetID(parts[1])
+
+	shareTarget, response, err := vpcClient.GetShareMountTargetWithContext(context, getShareMountTargetOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[DEBUG] GetShareMountTargetWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+
```

Comment:
Can we check shareTarget != nil and then access *shareTarget.X for eliminating nil pointer exception .
or Add in above 

if err != nil  || shareTarget = nil{

Reply:
added shareTarget == nil check